enhancements/network/dpu/overview.md

```diff
@@ -0,0 +1,239 @@
+  with extensive offload capabilities, but not necessarily with the
+  ability to run arbitrary user software. For example, the NVIDIA
+  Mellanox ConnectX-5 and ConnectX-6.
+
+- **DPU** / **Data Processing Unit** - a Smart NIC with a full CPU,
+  RAM, storage, etc, running a full operating system, able to offload
+  network processing and potentially other tasks from its host system.
+  For example, the NVIDIA BlueField-2.
+
+  (Some vendors other than NVIDIA are using the terms "Smart NIC" and
+  "DPU" slightly differently, but for now at least, this is the way we
+  are defining these terms.)
+
+- **Two-Cluster Design** - any architecture for deploying an OCP
+  cluster with (some) nodes containing DPUs, where the DPUs are nodes
+  in a second OCP cluster. (For purposes of this term, the
+  architecture still counts as "two-cluster" even if HyperShift is
+  involved and there are actually three clusters.)
+
+- **Infra Cluster** - in a Two-Cluster Design, the OCP cluster which
+  contains the DPUs as nodes. (It may also include master or
+  infrastructure nodes, depending on the particular details of the
+  two-cluster design.)
+
+- **Tenant Cluster** - in a Two-Cluster Design, the OCP cluster
+  containing the (presumably-x86) hosts which contain the DPUs. It may
+  also include nodes that do not have DPUs.
+
+## Motivation
+
+### User Stories
+
+#### Network Offload
+
+As a cluster administrator, I would like to offload network processing
+from the host CPU to the NIC, to increase throughput and to free up
+host CPU cycles for workloads rather than network processing.
+
+In particular:
+
+- We can move all of the node-side OVS, OVN, and OVN Kubernetes
+  daemons from the host CPU to the NIC CPU. (The host side would
+  retain only a CNI plugin binary.)
+
+- We can move IPsec encryption from the host CPU to the NIC CPU
+
+- Is there interest in moving DNS or routers off the host CPU?
+
+- ...
+
+#### Security Offload
+
+As a cluster administrator, I would like to use a DPU to monitor its
+host node. The NIC would run in an isolated mode, where the host OS
+has no control over it. Possibly even the cluster administrators of
+the host's OCP cluster would have no control over the NIC.
+
+#### Storage Offload
+
+(FIXME: Details? NVIDIA talks about this case in their marketing
+materials but I haven't heard as much about it in an OCP context and
+maybe this isn't an important case for us?)
+
+The BlueField-2 cards have M.2 NVMe ports and can have directly
+attached SSDs.
+
+### Goals
+
+- Design a system for provisioning and maintaining a RHEL-based
+  software stack on Smart NICs to support features such as "[Smart NIC
+  OVN Offload]".
+
+  - Although this enhancement is currently written to assume that
+    this means running OCP on the NICs, this is not yet cast in
+    stone. Some other approaches are discussed under "Alternatives"
+    and one of those could potentially replace the OCP approach.
```

Comment:
Do we really mean `replace` or they will co-exist in different setups? This sounds like a different path forward, would like to hear what is the other approach.

Reply:
I don't think we would want to support multiple very-different architectures. We're going to want to settle on one architecture that's flexible enough to handle all of our use cases.